catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/PipelineServiceRepository.java

```diff
@@ -0,0 +1,139 @@
+ *  contributor license agreements. See the NOTICE file distributed with
+ *  this work for additional information regarding copyright ownership.
+ *  The ASF licenses this file to You under the Apache License, Version 2.0
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License. You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.openmetadata.catalog.jdbi3;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.joda.time.Period;
+import org.joda.time.format.ISOPeriodFormat;
+import org.openmetadata.catalog.Entity;
+import org.openmetadata.catalog.entity.services.PipelineService;
+import org.openmetadata.catalog.exception.EntityNotFoundException;
+import org.openmetadata.catalog.type.Schedule;
+import org.openmetadata.catalog.util.EntityUtil;
+import org.openmetadata.catalog.util.JsonUtils;
+import org.skife.jdbi.v2.sqlobject.Bind;
+import org.skife.jdbi.v2.sqlobject.CreateSqlObject;
+import org.skife.jdbi.v2.sqlobject.SqlQuery;
+import org.skife.jdbi.v2.sqlobject.SqlUpdate;
+import org.skife.jdbi.v2.sqlobject.Transaction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+
+import static org.openmetadata.catalog.exception.CatalogExceptionMessage.entityNotFound;
+
+
+public abstract class PipelineServiceRepository {
+  private static final Logger LOG = LoggerFactory.getLogger(PipelineServiceRepository.class);
+
+  @CreateSqlObject
+  abstract PipelineServiceDAO pipelineServiceDAO();
+
+  @CreateSqlObject
+  abstract EntityRelationshipDAO relationshipDAO();
+
+  @Transaction
+  public List<PipelineService> list(String name) throws IOException {
+    return JsonUtils.readObjects(pipelineServiceDAO().list(name), PipelineService.class);
+  }
+
+  @Transaction
+  public PipelineService get(String id) throws IOException {
+    return EntityUtil.validate(id, pipelineServiceDAO().findById(id), PipelineService.class);
+  }
+
+  @Transaction
+  public PipelineService getByName(String name) throws IOException {
+    return EntityUtil.validate(name, pipelineServiceDAO().findByName(name), PipelineService.class);
+  }
+
+  @Transaction
+  public PipelineService create(PipelineService pipelineService) throws JsonProcessingException {
+    // Validate fields
+    validateIngestionSchedule(pipelineService.getIngestionSchedule());
+    pipelineServiceDAO().insert(JsonUtils.pojoToJson(pipelineService));
+    return pipelineService;
+  }
+
+  public PipelineService update(String id, String description, URI url,
```

Comment:
Needs @Transaction annotation?

Reply:
Thanks for the catch. Updated in the latest commit